### PR TITLE
Implement R^2 as a PEW metric

### PR DESCRIPTION
**Pull request by bfolie**
# Citrine Python PR

## Description 
Allow the user to request R^2 as an evaluator metric.

This is a draft until any necessary changes are made to the Javascript client, but we can fine-tune the language in the meantime.

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)

### Adherence to team decisions
- [X] I have added tests for 100% coverage
- [X] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [X] I have bumped the version in setup.py

### CODE REVIEW

File: tests/informatics/test_predictor_evaluation_metrics.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 @pytest.fixture(params=[
     (RMSE(), "rmse", "RMSE"),
+    (RSquared(), "r_squared", "R^2"),
```
Comment (kroenlein) [inline tests/informatics/test_predictor_evaluation_metrics.py:9]: This isn't just relevant to this addition, but oughtn't we be importing these parameter lists directly from predictor_evaluation_metrics.py?

File: src/citrine/informatics/predictor_evaluation_metrics.py [hunk 2]
```diff
@@ -82,6 +84,24 @@ def __str__(self):
...
+    This dimensionless metric is equal to 1 - (mean squared error / variance of data).
+    It is equal to the coefficient of determination calculated with respect to the line
+    `predicted = actual`, hence it is commonly referred to as R^2. But unlike R^2 from
+    ordinary linear regression, this metric can be negative.
+    """
+
+    typ = properties.String("type", default="RSquared", deserializable=False)
+
+    def __repr__(self):
+        return "r_squared"
```
Comment (maxhutch) [inline src/citrine/informatics/predictor_evaluation_metrics.py:99]: In the backend, I'm pretty sure this is "R^2":
https://github.com/CitrineInformatics/mithril/blob/master/src/main/scala/io/citrine/nucleus/ml/analysis/crossvalidation/evaluators/metrics/RegressionMetrics.scala#L43

**Review by maxhutch — CHANGES_REQUESTED** [review]
Approach looks good but I don't think they repr and key match.

File: src/citrine/informatics/predictor_evaluation_metrics.py (hunk 2 above)
Comment (bfolie) [reply src/citrine/informatics/predictor_evaluation_metrics.py:99]: Good point--I didn't actually realize that those keys in mithril became the keys of the "metrics" dictionary.

**Comment by bfolie** [comment]
It works! Here's a jenkins job running updated e2e tests (https://github.com/CitrineInformatics/nextgen-devkit/pull/808) using this version of citrine-python: https://jenkins.corp.citrine.io/job/Devkit-custom/96/

**Review by maxhutch — APPROVED** [review]

File: tests/informatics/test_predictor_evaluation_metrics.py (hunk 1 above)
Comment (bfolie) [reply tests/informatics/test_predictor_evaluation_metrics.py]: We could, but then we'd end up asserting `str(metric) == metric.__str__` :p

This is really just boilerplate to satisfy coverage :/

**Pull request merged**